### PR TITLE
Fix #106 #118: Show/Search and Delete branches and tags from dropdown

### LINKED ISSUE #106: browse and search existing tags
#### Feature request:
This is a decomposition of issue #40. The functionality required here should allow the user to:

1. browse all existing tags

2. search for a certain tag

### PR DESCRIPTION
**Pull request by AhmadAlsaif**
Jamie implemented a feature on top of existing codebase to separate tags and branches. Made the search input field work and refactored both the design and code.

## Testing:
1. Navigate to the branches lists
2. observe that branches and tags are on two different lists
3. search for a branch and a tag
4. delete tags from the list

59650007-6f6ba600-9139-11e9-90e7-e075441372a5

### CODE REVIEW

**Review by mdesco18 — CHANGES_REQUESTED** [review]
Rebased old copy of master

**Comment by Leon0315** [comment]
I tested delete branch, the remote branch was deleted successfully 
Screen Shot 2019-06-21 at 5 53 25 PM
However the branch is still shown in the drop down list and refresh won't help. When click on that branch it will check out to that branch but actually that branch doesn't exist anymore. 
Screen Shot 2019-06-21 at 5 54 15 PM

Not sure if needed to test delete branch. Delete tags, commit and push works fine

File: app/misc/repo.ts [hunk 1]
```diff
@@ -609,8 +692,18 @@ function openRepository() {
           .then(function () {
             refreshAll(repo);
           }, function (err) {
+<<<<<<< HEAD
```
Comment (ajlarivi) [inline app/misc/repo.ts]: is this an unresolved merge conflict?

**Comment by cchampernowne** [comment]
On linux, push/commit are working for issue106 (still buggy but that is preexisting). Pull is not detecting remote repo (likely not related to issues with this branch).
Tagging functionality appears to be working.

**Review by ajlarivi — CHANGES_REQUESTED** [review]
**tested the following functionality successfully on MacOS**
- Navigate to the branches lists
- observe that branches and tags are on two different lists
- search for a branch and a tag
- delete tags from the list

additionally, as much functionality of the rest of the application was tested without finding any issues. commit, push, adding tags, fetch, etc. all performed as expected. 

The only issue found was an unresolved merge conflict in the code? Not really sure what the deal is there. See individual comment on the modified files.

File: app/misc/repo.ts (hunk 1 above)
Comment (AhmadAlsaif) [reply app/misc/repo.ts]: Resolved

**Comment by AhmadAlsaif** [comment]
@Leon0315, delete branch was not listed in the issues we were working on. However, there's another issue for it https://github.com/uvicgse/project-virtual-team-2/issues/129

**Review by ajlarivi — DISMISSED** [review]
merge conflict resolved. Approving.

**Comment by AhmadAlsaif** [comment]
> On linux, push/commit are working for issue106 (still buggy but that is preexisting). Pull is not detecting remote repo (likely not related to issues with this branch).
> Tagging functionality appears to be working.

If it's working well on your machine, can you approve it please?

**Pull request merged**